core/windows_and_tabs/window_snap.py

```diff
@@ -316,13 +319,32 @@ def window_split_position(m) -> Dict[int, list[RelativeScreenPos]]:
 
 @mod.action_class
 class Actions:
-    def snap_window(position: RelativeScreenPos) -> None:
+    def snap_window(
+        position: RelativeScreenPos, window: Optional[Window] = None
+    ) -> None:
         """Move the active window to a specific position on its current screen, given a `RelativeScreenPos` object."""
-        _snap_window_helper(ui.active_window(), position)
+        if window is None:
+            window = ui.active_window()
+        _snap_window_helper(window, position)
 
-    def snap_window_to_position(position_name: str) -> None:
+    def snap_window_to_position(
+        position_name: str, window: Optional[Window] = None
+    ) -> None:
         """Move the active window to a specifically named position on its current screen, using a key from `_snap_positions`."""
-        actions.user.snap_window(_snap_positions[position_name])
+        position: Optional[RelativeScreenPos] = None
+        if position_name in _snap_positions:
+            position = _snap_positions[position_name]
+        else:
+            screaming_snake_position = actions.user.formatted_text(
+                position_name, "ALL_CAPS,SNAKE_CASE"
+            )
+            actions.user.deprecate_command(
+                "2024-12-02",
+                f"snap_window_to_position('{position_name}')",
+                f"snap_window_to_position('{screaming_snake_position}')",
+            )
+            position = _snap_positions[screaming_snake_position]
```

Comment:
This should probably be a three-clause if:

- If a valid identifier was passed (works)
- if a previous spoken form was passed that matches a valid identifier when converted (convert, show deprecation warning → works)
- Something else was passed (just return KeyError or something similar, don't show the deprecation warning)

Presently this is mixing 2 and 3 because you're assuming the converted key is always present here without checking first; I would separate 3 out. Otherwise if I pass a nonsense identifier, it will still try to offer the deprecation code conversion, assuming I passed a spoken form, which isn't quite right.

Reply:
Makes sense. I updated it.